src/test/java/com/rackspace/ceres/app/web/WriteControllerTest.java

```diff
@@ -0,0 +1,65 @@
+package com.rackspace.ceres.app.web;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import com.rackspace.ceres.app.config.AppProperties;
+import com.rackspace.ceres.app.model.Metric;
+import com.rackspace.ceres.app.model.PutResponse;
+import com.rackspace.ceres.app.services.DataWriteService;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.reactive.WebFluxTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.reactive.server.WebTestClient;
+import reactor.core.publisher.Flux;
+
+@ActiveProfiles("test")
+@WebFluxTest(WriteController.class)
+@Import(AppProperties.class)
+public class WriteControllerTest {
+
+  @MockBean
+  DataWriteService dataWriteService;
+
+  @Autowired
+  WebTestClient webTestClient;
+
+  @Test
+  public void testPutMetrics() {
```

Comment:
Also add a put "single metric" version. I'm hoping it's the same but `Mono.just(metric)` for the body.

Reply:
This is for single metric only.

Reply:
You're passing one metric, but it's via a Flux so I'm guessing it's encoding as an array with a single metric.

It might require using an explicit textual body to differentiate a test between a single object payload:

https://github.com/racker/ceres/blob/5bdb244c466509ac26769c59c18645375688432a/testing.http#L5-L15

vs an array payload

https://github.com/racker/ceres/blob/5bdb244c466509ac26769c59c18645375688432a/testing.http#L21-L56

Reply:
Added a test case where request is passed as List